tiledb/sm/serialization/query.cc

```diff
@@ -1070,7 +1070,7 @@ tdb_unique_ptr<ASTNode> condition_ast_from_capnp(
 
 Status condition_from_capnp(
     const capnp::Condition::Reader& condition_reader,
-    QueryCondition* const condition) {
+    QueryCondition& condition) {
```

Comment:
Let's make this return a QueryCondition object and unstatus the function.

Reply:
Done!